### ISSUE TITLE
Incorrect description for  	0 59 10 ? 1/2 MON#1 *

### ISSUE THREAD

**Issue by maheshneogi** [jmrozanec/cron-utils#126]
Actual result : at 10:59 every February months Monday 1 of every month

Expected Result : SOMETHING LIKE   "The  1st Monday of every two months at 10:59"

**Comment by jmrozanec** [comment]
@maheshneogi thank you for reporting this - the description proposed looks much better :) Can we add a test case for this? I will be glad to accept the pull request. Thanks!

**Comment by fante76** [comment]
It seems there is a problem with iterations syntax. I found same problem with:
Expression:	0 15/4 12 * * ?
Description:	every 4 minutes at 12 hour
Next start:	Tue Nov 08 12:15:00 CET 2016

"Next Start" is calculated using org.apache.logging.log4j.core.util.CronExpression

The original code is:

	try {
	    CronDescriptor cd = CronDescriptor.instance(Locale.ENGLISH);
	    CronParser parser = new CronParser (CronDefinitionBuilder.instanceDefinitionFor(CronType.QUARTZ));
	    final String cronExpression = "0 15/4 12 * * ?";
	    Cron c = parser.parse(cronExpression);
	    System.out.println("Expression:\t"+cronExpression);
	    System.out.println("Description:\t"+cd.describe(c));
	    CronExpression ce = new CronExpression(cronExpression);
	    System.out.println("Next start:\t"+ce.getNextValidTimeAfter(new Date()));
	} catch (Exception e) {
	    // TODO Auto-generated catch block
	    e.printStackTrace();
	}

**Comment by jmrozanec** [comment]
@fante76 I agree. Currently I would need more time to come back with a better way to get descriptions. If you have ideas / proposals, I will be glad to discuss.

**Comment by fante76** [comment]
As you, if I could catch some extra time I will get a look in to the code.

Il 09/nov/2016 10:57, "jmrozanec" <notifications@github.com> ha scritto:

**Comment by Naxos84** [comment]
So you are looking for something that is working like: https://www.freeformatter.com/cron-expression-generator-quartz.html ?

**Comment by jmrozanec** [comment]
@Naxos84 we are looking to get that quality level of descriptions, but with the flexibility that cron-utils can provide. We created a [specific milestone](https://github.com/jmrozanec/cron-utils/milestone/8) to tackle this improvement. Probably some major changes will be required. Any thoughts on this are appreciated!

**Comment by Naxos84** [comment]
Ok. I am going to think about this as soon as I am more comfortable with
the project.

Am 17.11.2017 13:43 schrieb "jmrozanec" <notifications@github.com>:
